packages/react-reconciler/src/SchedulingProfiler.js

```diff
@@ -194,9 +194,16 @@ export function markComponentSuspended(
       const id = getWakeableID(wakeable);
       const componentName = getComponentNameFromFiber(fiber) || 'Unknown';
       const phase = fiber.alternate === null ? 'mount' : 'update';
+
+      // Following the non-standard fn.displayName convention,
+      // frameworks like Relay may also annotate Promises with a displayName,
+      // describing what operation/data the thrown Promise is related to.
+      // When this is available we should pass it along to the Scheduling Profiler.
+      const displayName = (wakeable: any).displayName || '';
```

Comment:
Note that this convention was already established and used by frameworks like Relay.

Note that we're adding the new value in a backwards-compatible way. (Older profiling data sets will just fallback to showing the component name.)